### DISCUSSION TITLE
How to extract trade buying price?

### DISCUSSION THREAD

**Discussion by Chunger-Lo** [kernc/backtesting.py#326] [Q&A]
Hi, I am a rookie to quantitative trades,

I really love the backtest output of this framework, 
and I just want to modify the tutorial example:

Instead of comparing long and short sma signal, 
my buying criteria is : current_price > ema
my selling criteria is : current_price < buy_price * 0.9
so I am wondering how to extract the buy_price (the price when I bought) from the trade?
and how to use it as a selling criteria

Sorry for the dumb question, please let me if there's any misunderstanding.
Thank you!

```
from backtesting import Backtest, Strategy
from backtesting.lib import crossover
from backtesting.test import SMA, GOOG
import pandas as pd
import talib

def EMA(arr: pd.Series, n: int) -> pd.Series:
    res = talib.EMA(pd.Series(arr), timeperiod = n)
    return res

class myCross(Strategy):
    ema_n = 144
    def init(self):
        price = self.data.Close
        self.ema = self.I(EMA, price, self.ema_n)
    
    def next(self):
        if crossover(self.data.Close, self.ema):
            self.buy()
        elif crossover(self.ema, self.trades[-1]['entry_price'] * 0.9):
            self.sell()
            
bt = Backtest(data = GOOG, strategy = myCross, commission=.002,
              exclusive_orders=True)
stats = bt.run()
bt.plot()
```

**Comment by kernc** [comment]
Did you see [`Strategy.trades`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.trades) and [`Trade.entry_price`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Trade.entry_price).

Something like:
```py
        for trade in self.trades:
            if self.data.Close < trade.entry_price * .9:
                trade.close()
```

**Reply by Chunger-Lo** [reply]
It worked! thanks for your help.
